### PR TITLE
Fix #4390: Sort REPL autocomplete suggestions

### LINKED ISSUE #4390: REPL completion output is not sorted
image

### PR DESCRIPTION
**Pull request by ankitson**
This seems to be a quick fix for https://github.com/lampepfl/dotty/issues/4390. Its my first PR to dotty so I might be missing something.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/interactive/Interactive.scala [hunk 1]
```diff
@@ -219,7 +219,7 @@ object Interactive {
       case _  => getScopeCompletions(ctx)
     }
     interactiv.println(i"completion with pos = $pos, prefix = $prefix, termOnly = $termOnly, typeOnly = $typeOnly = ${completions.toList}%, %")
-    (completionPos, completions.toList)
+    (completionPos, completions.toList.sortBy(_.name.show))
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/interactive/Interactive.scala]: You can use `_.name.toString`

**Comment by ankitson** [comment]
Done!

File: compiler/test/dotty/tools/repl/TabcompleteTests.scala [hunk 2]
```diff
@@ -65,4 +65,14 @@ class TabcompleteTests extends ReplTest {
       List("\"", ")", "'", "¨", "£", ":", ",", ";", "@", "}", "[", "]")
         .foreach(src => assertTrue(tabComplete(src).suggestions.isEmpty))
     }
+
+  @Test def sortedCompletions: Unit =
+    fromInitialState { implicit state  =>
+      val src = """class Foo { def comp1 = 1; def compa = 3; def compA = 4 }"""
```
Comment (allanrenucci) [inline compiler/test/dotty/tools/repl/TabcompleteTests.scala]: Use simple quotes. Triple quotes are not needed here.

Comment (allanrenucci) [reply compiler/test/dotty/tools/repl/TabcompleteTests.scala]: I prefer the code snippet I gave you. I think the ordering is clearer:
```scala
class Foo { def comp3 = 3; def comp1 = 1; def comp2 = 2 }
```
```scala
val expected = List("comp1", "comp2", "comp3")
```

File: compiler/src/dotty/tools/dotc/interactive/Interactive.scala [hunk 3]
```diff
@@ -219,7 +219,7 @@ object Interactive {
       case _  => getScopeCompletions(ctx)
     }
     interactiv.println(i"completion with pos = $pos, prefix = $prefix, termOnly = $termOnly, typeOnly = $typeOnly = ${completions.toList}%, %")
-    (completionPos, completions.toList)
+    (completionPos, completions.toList.sortBy(_.name.toString))
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/interactive/Interactive.scala]: @smarter We have an ordering for `Name` that differs from the default `String` ordering. I am wondering if we should use it here? The Name ordering will not group type names and term names together

Comment (smarter) [reply compiler/src/dotty/tools/dotc/interactive/Interactive.scala]: I think they should be grouped together, but we should also not display type completions if we're not in a type context

Comment (smarter) [reply compiler/src/dotty/tools/dotc/interactive/Interactive.scala]: I think they should be grouped together, but we should also not display type completions if we're not in a type context

Comment (allanrenucci) [reply compiler/src/dotty/tools/dotc/interactive/Interactive.scala]: Sure but that's probably out of the scope of this PR

**Pull request merged**